src/gt4py/storage/storage.py

```diff
@@ -294,10 +298,14 @@ def copy(self):
     def gpu_view(self):
         return storage_utils.gpu_view(self)
 
+    @property
+    def data(self):
+        return cp.asarray(super().data)
+
```

Comment:
This would give you the data attribute of a newly allocated GPU buffer, not the one of the already allocated buffer, since you pass the MemoryView of a CPU buffer, which cupy will then copy. I think what you want here is  
```suggestion
    @property
    def data(self):
        return cp.asarray(self)
```

Reply:
Thank you @gronerl, I will try that.

Reply:
Hello Linus,

The suggested change works well for the existing gt4py tests. However, we have cases in our dycore code that throw the error:
```python
E   cupy.cuda.runtime.CUDARuntimeError: cudaErrorInvalidValue: invalid argument
```
I have added a test, `test_storage.test_add_gpu_managed`, that replicates the error.

Based on my reading and debugging, the storage resulting from the addition, `q3`, is not produced from the constructor, but rather from the `__array_finalize__` method. I believe the solution is probably to override that method in the `GPUStorage` class, and update the object to prevent the error, e.g,
```python
def __array_finalize__(self, obj):
    super().__array_finalize__(self, obj)
   # Correct the storage object...
```

Unfortunately, I have not yet been able to determine what those corrections might be. Do you have any suggestions? Thank you!